packages/components/src/components/domainproperties/dataclasses/DataClassPropertiesPanel.tsx

```diff
@@ -97,10 +101,17 @@ class DataClassPropertiesPanelImpl extends React.Component<Props, State> {
         updateDomainPanelClassList(prevProps.useTheme, undefined, PROPERTIES_HEADER_ID);
     }
 
-    setIsValid() {
+    setIsValid(newModel?: DataClassModel) {
         const { model, onChange } = this.props;
-        const isValid = model && model.hasValidProperties();
-        this.setState(() => ({isValid}), () => onChange(model));
+        const updatedModel = newModel || model;
+        const isValid = updatedModel && updatedModel.hasValidProperties();
+        this.setState(() => ({isValid}),
+            () => {
+                // Issue 39918: only consider the model changed if there is a newModel param
+                if (newModel) {
+                    onChange(updatedModel)
```

Comment:
Not sure I like this here, particularly given the name of the method. Feels side-effecty, took me a bit to figure out this is where the model state is getting passed up the stack.

Reply:
Yeah, I agree. See my comments from the same section of the ListPropertiesPanel component. I'd prefer to keep this in sync for now and then refactor them together in the next round of polish/updates to the designers (which I'm hoping to do after the SampleTypeDesigner component and this one are merged to master).